code/drasil-gool/lib/Drasil/GOOL/LanguageRenderer/CppRenderer.hs

```diff
@@ -1246,14 +1261,7 @@ instance MathConstant CppSrcCode where
     addMathHImport (mkStateVal double cppPi)
 
 instance VariableValue CppSrcCode where
-  valueOf vr' = do
-    vr <- vr'
-    its <- zoom lensVStoMS getIter
-    let namevr = variableName vr
-    if namevr `elem` its then
-      mkStateVal (iterator $ toState $ variableType vr)
-        (parens $ cppDeref <> RC.variable vr)
-      else G.valueOf vr'
+  valueOf = G.valueOf
```

Comment:
I'm just curious, what was the reason for this change?  To be honest I don't exactly know what the original code does. 
 It looks like it's checking for some edge case before calling `G.valueOf`.  Is it safe to get rid of the check?

Reply:
When I tried to use this it assumed the variable I wanted the valueOf was a pointer and this was causing issues so I changed it.

Reply:
From what I can tell, I believe you're right that we can safely remove that line.  It looks like you changed the C++ implementation of `forEach` (for the better imo) so that the check isn't needed.

I noticed though, with that change the `getIter` function isn't used at all across GOOL.  This leads me to believe that we can remove completely the `_iterators` field in [State.hs](https://github.com/JacquesCarette/Drasil/blob/8ca26be6df6d3f4c24627e76d051637b0ff85b16/code/drasil-gool/lib/Drasil/GOOL/State.hs#L124).  It's only used by C++, which doesn't seem to need it anymore.

That change might be somewhat large, so it might be better to look into it after this PR is merged.

Reply:
That seems like a good way forward - could either @NoahCardoso or @B-rando1 write an issue for this, so it doesn't get lost?

Reply:
Done!